fastchat/model/compression.py

```diff
@@ -167,12 +167,18 @@ def load_compress_model(model_path, device, torch_dtype, use_fast, revision="mai
         tmp_state_dict = torch.load(filename, map_location=lambda storage, loc: storage)
         for name in tmp_state_dict:
             if name in linear_weights:
-                tensor = tmp_state_dict[name].to(device).data.to(torch_dtype)
```

Comment:
Could you try something like this instead of if/else?
```tensor = tmp_state_dict[name].to(torch_dtype).to(device).data```

If it works, apply the same change for L178.

Reply:
I just fix it by inline `to` and remove `mps` condition.It works for chatglm2-6b:
<img width="1286" alt="image" src="https://github.com/lm-sys/FastChat/assets/8603593/227e2bce-a1b0-49b1-b9a6-9ecc33cd4583">